MetaMask/metamask-extension
# fix: Fix gas fee polling not being stopped for popup UI instances in chrome

## **Description**

Currently extension popups in chrome do not properly clean up react effects. Previously, gas fee polling started from UI components were also tracked by MetaMaskController and cleaned up when the controller lost connection with the popup (or fullscreen, or notification) UI instances. The new gas fee polling hook failed to account for this and did not properly track its polling tokens with background which introduced the possibility of gas fee polling not stopping correctly in certain instances. This PR fixes this by updating the `usePolling` hook to add and remove poll tokens from app state and also updates MetaMaskController to stop any active gas fee polling based on the polling tokens in app state.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23594?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2014

## **Manual testing steps**

In chrome

1. Open a popup instance of extension (click the extension icon from the toolbar)
2. Open background console network tab
3. Get to the send transaction confirmation screen
4. Verify gas fee requests are happening
5. Close the popup instance by unfocusing it
6. Verify gas fee requests have stopped

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
